modules/apps/asset/asset-categories-service/src/main/java/com/liferay/asset/categories/internal/service/messaging/AssetCategoriesReindexConfiguration.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.asset.categories.internal.service.messaging;
+
+import com.liferay.portal.kernel.messaging.Destination;
+import com.liferay.portal.kernel.messaging.DestinationConfiguration;
+import com.liferay.portal.kernel.messaging.DestinationFactory;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.util.MapUtil;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Manuel Rives
+ */
+@Component(service = {})
+public class AssetCategoriesReindexConfiguration {
```

Comment:
Why do we need this class? couldnt we add this to AssetCategoriesReindexMessageListener activate method?  see com.liferay.document.library.preview.audio.internal.messaging.AudioProcessorMessageListener#activate as example

Reply:
I see both patterns, 
- creation of an independent class and listener (AntivirusAsyncMessagingConfigurator /& AntivirusAsyncMessageListener)
- put on the listener (CTScoreMessageListener)

So I'm doubting about this.

Reply:
Having that in a single class has less memory footprint